src/orion/executor/single_backend.py

```diff
@@ -5,8 +5,43 @@
 
 """
 import functools
+import traceback
 
-from orion.executor.base import BaseExecutor
+from orion.executor.base import AsyncException, AsyncResult, BaseExecutor
+
+
+class _Future:
+    """Wraps a partial function to act as a Future"""
+
+    def __init__(self, future):
+        self.future = future
+        self.result = None
+        self.exception = None
+
+    def get(self, timeout=None):
+        if self.result:
+            return self.result
+
+        if self.exception:
+            raise self.exception
+
+        self.result = self.future()
+        return self.result
+
+    def wait(self, timeout=None):
+        try:
+            self.result = self.future()
+        except Exception as e:
+            self.exception = e
+
+    def ready(self):
+        return self.result is None
+
+    def successful(self):
```

Comment:
This function should be documented at a base future class. I would not expect it to raise ValueError is the future is not done.

Reply:
added